main.py

```diff
@@ -41,12 +43,14 @@ def refresh_repos():
             break
         except subprocess.CalledProcessError as e:
             err = e
+            error_occurred = True
```

Comment:
The error_occured and update_occured are global variables, so you need to define them as global in the function, otherwise the assignment applies to local scope and the global variable is not assigned.

https://docs.python.org/3/tutorial/classes.html#python-scopes-and-namespaces